### DISCUSSION TITLE
Is there any kind of hook for server reload in dev mode?

### DISCUSSION THREAD

**Discussion by davnicwil** [remix-run/remix#1160] [Q&A]
Is there any way to hook into when the (node) server hot reloads on save in dev mode?

I'd like to do things like clean up database clients etc when the server hot reloads - after a certain number of hot reloads I see errors like connection / client limits exceeded and have to hard restart the server to continue.

**Comment by Girish21** [comment]
to work around this you can follow the steps described here [connect to the database](https://remix.run/docs/en/v1/tutorials/jokes#connect-to-the-database). This will stop creating new connections on every hot reload and will reuse the existing connection in the development environment
